### PR TITLE
[Community pipeline] Add github mechanism

### PR DESCRIPTION
**Pull request by patrickvonplaten**
Community pipelines can now be versioned with a `custom_revision` input to `from_pretrained`. If nothing is specified it will default to what is currently installed locally.

### CODE REVIEW

File: tests/test_pipelines.py [hunk 1]
```diff
@@ -207,6 +207,15 @@ def test_load_custom_pipeline(self):
         # under https://huggingface.co/hf-internal-testing/diffusers-dummy-pipeline/blob/main/pipeline.py#L24
         assert pipeline.__class__.__name__ == "CustomPipeline"
 
+    def test_load_custom_github(self):
+        pipeline = DiffusionPipeline.from_pretrained(
+            "google/ddpm-cifar10-32", custom_pipeline="one_step_unet", custom_revision="0.10.2"
+        )
+        pipeline = pipeline.to(torch_device)
+        # NOTE that `"CustomPipeline"` is not a class that is defined in this library, but solely on the Hub
+        # under https://huggingface.co/hf-internal-testing/diffusers-dummy-pipeline/blob/main/pipeline.py#L24
```
Comment (pcuenca) [inline tests/test_pipelines.py:216]:
```suggestion
```

File: tests/test_pipelines.py [hunk 2]
```diff
@@ -207,6 +207,15 @@ def test_load_custom_pipeline(self):
         # under https://huggingface.co/hf-internal-testing/diffusers-dummy-pipeline/blob/main/pipeline.py#L24
         assert pipeline.__class__.__name__ == "CustomPipeline"
 
+    def test_load_custom_github(self):
+        pipeline = DiffusionPipeline.from_pretrained(
+            "google/ddpm-cifar10-32", custom_pipeline="one_step_unet", custom_revision="0.10.2"
```
Comment (pcuenca) [inline tests/test_pipelines.py:212]: How does this test verify that the pipeline came from `0.10.2`? We could, for example:
- Download from a version where the pipeline did not exist (should fail).
- Download from the first version where it does (succeeds).

**Review by pcuenca — APPROVED** [review]
Very cool!

File: tests/test_pipelines.py (hunk 2 above)
Comment (patrickvonplaten) [reply tests/test_pipelines.py:212]: Very good point! Will adapt the test :-)

**Pull request merged**